File: ts/src/values/nullValue.ts
import { CompareResult, IValue } from "./ivalues";

export class NullValue implements IValue
{
    public static readonly Value = new NullValue();

    public typename() { return 'null'; }
    public toString() { return 'null'; }

    public compareTo(other: IValue): CompareResult
    {
        if (!(other instanceof NullValue))
        {
            return 1;
        }

        return 0;
    }
}

File: unity/simple-stack-vm-unity/Assets/Scripts/SimpleStackVM/Values/NullValue.cs
#nullable enable

namespace SimpleStackVM
{
    public struct NullValue : IValue
    {
        #region Fields
        public static readonly NullValue Value = new NullValue();
        public string TypeName => "null";
        #endregion

        #region Methods
        public int CompareTo(IValue? other) => (other == null || other is NullValue) ? 0 : 1;
        public override string ToString() => this.TypeName;
        #endregion
    }
}